python/swe/README.md

```diff
@@ -1,50 +1,84 @@
-# README for swe.py
+# SWE Development Kit
 
 ## Overview
 
-The `swe.py` script is part of the Composio software engineering (SWE) agent framework.
-It is designed to automate tasks related to software development, including issue resolution, code reviews, and patch submissions using AI-driven agents.
+Composio-swe is a powerful framework for building and evaluating software engineering agents. With this toolkit, you can:
+
+- Build your own custom agents
+- Utilize pre-built agents provided by Composio
+- Run the SWE-Bench benchmark to evaluate agent performance
 
 ## Dependencies
 
-1. Docker Desktop should be installed.
-2. Get the Github Access Token.
-3. Install the dependencies using `pip install -r requirements.txt`.
-4. Add the LLM configuration via `OPENAI_API_KEY` or `ANTHROPIC_API_KEY` or (`AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_ENDPOINT`) environment variables.
-5. Add the environment variable `export GITHUB_ACCESS_TOKEN = <git_access_token>`.
-6. If you want to use Helicone, add the environment variable `export HELICONE_API_KEY = <helicone_api_key>`.
+Before getting started, ensure you have the following set up:
+
+1. **GitHub Access Token**:
+
+   - Create one at https://github.com/settings/tokens with necessary permissions
+   - Set the environment variable: `export GITHUB_ACCESS_TOKEN=<your_token>`
+
+2. **Installation**:
+
+   ```
+   pip install composio-swe composio-core
+   ```
+
+3. **Optional CrewAI Support**:
+   If you plan to use CrewAI for your agent:
+
+   ```
+   pip install crewai composio-crewai
+   ```
+
+4. **LLM Configuration**:
+   - Set up OpenAI API key: `export OPENAI_API_KEY=<your_key>`
+   - Other LLMs can be configured by modifying the code (details provided later)
+
+## Getting Started
+
+### Scaffolding a New Agent
+
+1. Run the following command:
+
+   ```
+   composio-swe scaffold crewai -o <path>
+   ```
+
+   This creates a new agent in `<path>/agent` with four key files:
+
+   - `main.py`: Entry point to run the agent on your issue
+   - `agent.py`: Core agentic logic (edit this to customize behavior)
+   - `prompts.py`: Agent prompts
+   - `benchmark.py`: SWE-Bench benchmark runner
+
+   > **Note**: By default, the SWE-agent runs in Docker. To run locally, modify the `workspace_env` in `agent/agent.py`.
+
+2. Run the agent:
+
+   ```
+   python agent/main.py
+   ```
 
-## Usage
+   You'll be prompted for the repository name and issue.
 
-To change the script quickly:
+3. Run the benchmark:
+   ```
+   python agent/benchmark.py
+   ```
 
-1. Change the issue_config in swe_run.py
-2. Run the script with `python swe_run.py`
+## Extending Functionality
 
-To modify the agent and improve the agent's performance:
+### Adding a New Tool
 
-1. Modify the agent's code in swe.py
-2. Run the script with `python swe_run.py`
+To add a new local tool for use within the agent, refer to the [Composio SDK documentation](https://docs.composio.dev/sdk/python/local_tools).
```

Comment:
nit: thsi local tool page opens in the same browser-tab, if its a small thing, we should open it in new tab